### ISSUE TITLE
haproxy 2.2-dev8 log regression

### ISSUE THREAD

**Issue by khvzak** [haproxy/haproxy#654]
## Output of `haproxy -vv` and `uname -a`
```
HA-Proxy version 2.2-dev8 2020/05/22 - https://haproxy.org/
Status: development branch - not safe for use in production.
Known bugs: https://github.com/haproxy/haproxy/issues?q=is:issue+is:open
Running on: Darwin 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
Build options :
  TARGET  = osx
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-ignored-qualifiers -Wno-missing-field-initializers -Wno-implicit-fallthrough -Wno-string-plus-int -Wtype-limits -Wshift-negative-value -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_THREAD=1 USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1

Feature list : -EPOLL +KQUEUE -NETFILTER -PCRE -PCRE_JIT +PCRE2 -PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED -BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY -LINUX_TPROXY -LINUX_SPLICE +LIBCRYPT -CRYPT_H -GETADDRINFO +OPENSSL +LUA -FUTEX -ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY -TFO -NS -DL -RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER -PRCTL -THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=1).
Built with OpenSSL version : OpenSSL 1.1.1g  21 Apr 2020
Running on OpenSSL version : OpenSSL 1.1.1g  21 Apr 2020
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.5
Built with clang compiler version 11.0.3 (clang-1103.0.32.62)
Built with transparent proxy support using:
Built with PCRE2 version : 10.35 2020-05-09
PCRE2 library supports JIT : no (USE_PCRE2_JIT not set)
Encrypted password support via crypt(3): yes
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")

Available polling systems :
     kqueue : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use kqueue.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2
            fcgi : mode=HTTP       side=BE        mux=FCGI
       <default> : mode=HTTP       side=FE|BE     mux=H1
       <default> : mode=TCP        side=FE|BE     mux=PASS

Available services : none

Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [TRACE] trace
        [COMP] compression
```

```
Darwin C02V37K2HTD8 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```

## What's the configuration?
```
global
    master-worker
    log "127.0.0.1:1514" len 16384 format rfc3164 daemon info
    log-send-hostname

defaults
    mode http
    log global
    log-format "${HAPROXY_JSON_LOG_FORMAT}"

listen http-in
    bind 127.0.0.1:8080
    http-request return status 200
```

## Steps to reproduce the behavior
Run:
```
$ HAPROXY_JSON_LOG_FORMAT="{}" haproxy -W -d -f haproxy.cfg
```
And then
```
$ curl -v http://localhost:8080/
```

## Actual behavior

Bytes received from the UDP socket listening on port 1514 would be:
```
<30>May 27 15:51:29 C02V37K2HTD8 haproxy[59800]: {}
bytes: [60, 51, 48, 62, 77, 97, 121, 32, 50, 55, 32, 49, 53, 58, 53, 49, 58, 50, 57, 32, 67, 48, 50, 86, 51, 55, 75, 50, 72, 84, 68, 56, 32, 104, 97, 112, 114, 111, 120, 121, 91, 53, 57, 56, 48, 48, 93, 58, 32, 123, 125, 0, 10]
```
The last bytes are: `[0, 10]`
`0` is wrong there. This causes syslog to escape this byte to `#000`

## Expected behavior

On haproxy 2.2-dev7 there is no `0` byte.

## Do you have any idea what may have caused this?

Could be http://git.haproxy.org/?p=haproxy.git;a=commit;h=9e8ea0ae6f63fe1a8119d99b1218ccc095b390ac

**Comment by wtarreau** [comment]
Possible indeed. I've pinged @EmericBr about it, especially since this fix is waiting in 2.1 as well.

**Comment by wtarreau** [comment]
I've just merged Emeric's fix and backported it to 2.1 which should be issued imminently. I'm thus marking it fixed and closing since there's no more version affected. Do no hesitate to reopen if you face another issue around this, but I'm quite confident that we've finally nailed this very old issue down now that logs are properly normalized early in the function.
